pkg/common-controller/snapshot_controller.go

```diff
@@ -754,17 +759,17 @@ func (ctrl *csiSnapshotCommonController) storeContentUpdate(content interface{})
 	return utils.StoreObjectUpdate(ctrl.contentStore, content, "content")
 }
 
-// updateSnapshotStatusWithEvent saves new snapshot.Status to API server and emits
+// updateSnapshotErrorStatusWithEvent saves new snapshot.Status to API server and emits
 // given event on the snapshot. It saves the status and emits the event only when
 // the status has actually changed from the version saved in API server.
 // Parameters:
 //   snapshot - snapshot to update
 //   eventtype, reason, message - event to send, see EventRecorder.Event()
-func (ctrl *csiSnapshotCommonController) updateSnapshotErrorStatusWithEvent(snapshot *crdv1.VolumeSnapshot, eventtype, reason, message string) error {
-	klog.V(5).Infof("updateSnapshotStatusWithEvent[%s]", utils.SnapshotKey(snapshot))
+func (ctrl *csiSnapshotCommonController) updateSnapshotErrorStatusWithEvent(snapshot *crdv1.VolumeSnapshot, ready *bool, eventtype, reason, message string) error {
```

Comment:
logic lgtm
I have a small suggestion on implementation.
Passing in a pointer of bool here seems to be a bit misleading, i.e., caller can actually can pass in a "true" into this call.

How about taking in a bool parameter, say "unreadySnapshot", this way its a bit more clear?

Please also update the parameters in comments.

Reply:
Sure.  I named the bool parameter "setReadyToFalse" to be more explicit.  Addressed your comments.  PTAL.